core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableInterpretable.java

```diff
@@ -135,7 +135,8 @@ static Bindable getBindable(ClassDeclaration expr, String s, int fieldCount)
       throws CompileException, IOException, ExecutionException {
     ICompilerFactory compilerFactory;
     try {
-      compilerFactory = CompilerFactoryFactory.getDefaultCompilerFactory();
+      compilerFactory = CompilerFactoryFactory.getDefaultCompilerFactory(
```

Comment:
Why do we need to pass explicitly the class loader of the class? Is this change necessary due to the Janino update or we are trying to tackle another problem at the same time?

Reply:
@zabetak Good question. 

The root issue is that using Calcite in an "uber" JAR fails. When Calcite tries to load the Janino compiler using the `CompilerFactoryFactory`, it fails in the previous version of Janino commons library. This is because It was using the `Thread.currentThread().getContextClassLoader()` which was `null` at runtime.

I worked with the team at Janino to find a solution - where Calcite can pass its class loader to the Janino factory class so that it has the working class's class loader.

So we needed the latest version of Janino to get a fix to the problem of using Calcite in an "uber" JAR. 

Please see the origin [JIRA  issue](https://issues.apache.org/jira/browse/CALCITE-3745).

Reply:
Got it, thanks for the explanation.